### DISCUSSION TITLE
Location não encontrada, devo retornar 400 ou 404?

### DISCUSSION THREAD

**Discussion by mvmarcal** [bacen/pix-api#337] [Q&A]
Identifiquei que no Swagger temos o 2 Status sendo utilizados para Location não encontrada:
- Status 404 no **PayloadLocationNaoEncontrado** quando a Location não é encontrada via [GET]/loc/{id}
- Status 400 no **PayloadLocationConsultaInvalida** com a violação específica do endpoint GET /loc/{id} utilizando a seguinte mensagem: "{id} não identifica uma location existente"

Gostaria de saber qual destas devo utilizar?

Abaixo o trecho do Swagger:
> ### Tag PayloadLocation
> Esta seção reúne erros referentes a endpoints que tratam do gerenciamento de locations.
> **PayloadLocationNaoEncontrado**
> 
> - Significado: Location não encontrada para o id informado.
> - HTTP Status Code: 404.
> - _endpoints: [GET|PATCH] /loc/{id}, DELETE /loc/{id}/txid._
> 
> **PayloadLocationOperacaoInvalida**
> 
> - Significado: a presente requisição busca criar uma location sem respeitar o schema estabelecido.
> - HTTP Status Code: 400.
> - endpoints: POST /loc.
> 
> Violações para o endpoint POST /loc:
> - o campo tipoCob não respeita o schema.
> 
> **PayloadLocationConsultaInvalida**
> - Significado: os parâmetros de consulta à lista de locations não respeitam o schema ou não fazem sentido semanticamente.
> - HTTP Status Code: 400.
> - endpoints: GET /loc e GET /loc/{id}.
> 
> Violações específicas para o endpoint GET /loc:
> - algum dos parâmetros informados para a consulta não respeitam o schema.
> - o timestamp representado pelo parâmetro fim é anterior ao timestamp representado pelo parâmetro inicio.
> - o parâmetro paginacao.paginaAtual é negativo.
> - o parâmetro paginacao.itensPorPagina é negativo.
> 
> _Violações específicas para o endpoint GET /loc/{id}:_
> - _{id} não identifica uma location existente_

**Comment by ninrod** [comment]
404: objeto não existe. 
400: objeto existe, mas você passou parâmetros inválidos para a consulta do objeto, que existe.

**Reply by mvmarcal** [reply]
Mas no caso seguinte, não existe uma Location com o ID especificado.

> Violações específicas para o endpoint GET /loc/{id}:
> {id} não identifica uma location existente

**Reply by ninrod** [reply]
```
Violações específicas para o endpoint GET /loc/{id}:

{id} não identifica uma location existente
``` 

Aí já é um bug. Esse problema já consta aqui, então não precisaria repetí-lo, realmente. 

![image](https://user-images.githubusercontent.com/8352747/109533773-59cb1b00-7a99-11eb-99d6-c24ffa3ecb37.png)

@mvmarcal pode converter como bug? vou marcar e tentar encaixar na `2.2.2`, obrigado.

**Reply by ninrod** [reply]
corrigido.

**Reply by mvmarcal** [reply]
Obrigado pela correção, acabei não convertendo em Bug, desculpe.
